pkg/provider/aci_init_container_test.go

```diff
@@ -0,0 +1,201 @@
+package provider
+
+
+import (
+	"context"
+	"testing"
+
+	"github.com/virtual-kubelet/virtual-kubelet/errdefs"
+	"github.com/golang/mock/gomock"
+	"github.com/virtual-kubelet/azure-aci/pkg/client"
+	"github.com/virtual-kubelet/node-cli/manager"
+	"gotest.tools/assert"
+	is "gotest.tools/assert/cmp"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func TestCreatePodWithInitContainers(t *testing.T) {
+
+	initContainerName1 := "init-container-1"
+	initContainerName2 := "init-container-2"
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	aciMocks := createNewACIMock()
+	aciMocks.MockCreateContainerGroup = func(ctx context.Context, resourceGroup, podNS, podName string, cg *client.ContainerGroupWrapper) error {
+		containers := *cg.ContainerGroupPropertiesWrapper.ContainerGroupProperties.Containers
+		initContainers := *cg.ContainerGroupPropertiesWrapper.ContainerGroupProperties.InitContainers
+		assert.Check(t, cg != nil, "Container group is nil")
+		assert.Check(t, containers != nil, "Containers should not be nil")
+		assert.Check(t, initContainers != nil, "Container group is nil")
+		assert.Check(t, is.Equal(len(containers), 1), "1 Container is expected")
+		assert.Check(t, is.Equal(len(initContainers), 2), "2 init containers are expected")
+		assert.Check(t, initContainers[0].VolumeMounts != nil, "Volume mount should be present")
+		assert.Check(t, initContainers[0].EnvironmentVariables != nil, "Volume mount should be present")
+		assert.Check(t, initContainers[0].Command != nil, "Command mount should be present")
+		assert.Check(t, initContainers[0].Image != nil, "Image should be present")
+		assert.Check(t, *initContainers[0].Name == initContainerName1, "Name should be correct")
+		assert.Check(t, *initContainers[1].Name == initContainerName2, "Name should be correct")
+
+		return nil
+	}
+
+
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: podNamespace,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name: "container name",
+				},
+			},
+		},
+	}
+	cases := []struct {
```

Comment:
May we add test cases for init container with other containers. Also, it would be great if we add init container to one of the volumes unit tests.

Reply:
2 containers are present in the Pod definition along with different types of initContainers, for all test cases